cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -889,11 +879,11 @@ func (s *Supervisor) Shutdown() {
 		}
 	}
 
+	s.supervisorWG.Wait()
+
 	if s.healthCheckTicker != nil {
 		s.healthCheckTicker.Stop()
 	}
```

Comment:
This was another race condition, the health check ticker is recreated by the agent command goroutine, so we need to make sure that goroutine is exited before we try to stop it (otherwise we may leak a non-stopped ticker)